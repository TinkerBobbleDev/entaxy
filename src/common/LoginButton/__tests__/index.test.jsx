import React from 'react'
import renderer from 'react-test-renderer'
import { LoginButtonComponent } from '../'

describe('LoginButton', () => {
  const mochHandleLogin = jest.fn()
  const mochHandleLogout = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot with logged in user profile', () => {
    const component = renderer.create((
      <LoginButtonComponent
        user={{ isAuthenticatedWith: 'blockstack', name: 'Test name', username: 'Test' }}
        handleLogin={mochHandleLogin}
        handleLogout={mochHandleLogout}
        classes={{ }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
