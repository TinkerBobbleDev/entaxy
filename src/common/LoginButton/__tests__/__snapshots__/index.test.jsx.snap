// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton matches snapshot with logged in user profile 1`] = `
<div>
  <div
    aria-describedby={null}
    className="MuiAvatar-root-9 MuiAvatar-colorDefault-10"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="Test"
  />
  <button
    aria-owns={null}
    className="MuiButtonBase-root-38 MuiButton-root-12 MuiButton-text-14 MuiButton-flat-17 MuiButton-colorInherit-33"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-13"
    >
      Test name
    </span>
    <span
      className="MuiTouchRipple-root-41"
    />
  </button>
</div>
`;
